### PR TITLE
Update casadm -P implementation for standby mode

### PR DESCRIPTION
**Pull request by arutk**
Updated OCF returns error from cache/core get_stats() functions for standby cache. casadm shouldn't attempt to call stats IOCTL in this case and just present user with cache info.

### CODE REVIEW

File: casadm/statistics_model.c [hunk 1]
```diff
@@ -632,20 +632,26 @@ static int cache_stats(int ctrl_fd, const struct kcas_cache_info *cache_info,
...
+	bool standby;
+
 	cache_stats.cache_id = cache_id;
 	cache_stats.core_id = OCF_CORE_ID_INVALID;
 	cache_stats.part_id = OCF_IO_CLASS_INVALID;
 
-	if (ioctl(ctrl_fd, KCAS_IOCTL_GET_STATS, &cache_stats) < 0)
+	standby = !!(cache_info->info.state & (1 << ocf_cache_state_standby));
+
+	if (!standby && ioctl(ctrl_fd, KCAS_IOCTL_GET_STATS, &cache_stats)) {
```
Comment (robertbaldyga) [inline casadm/statistics_model.c:643]: Nested `if()` would be more clear.

File: casadm/statistics_model.c [hunk 2]
```diff
@@ -770,6 +776,17 @@ int cache_status(unsigned int cache_id, unsigned int core_id, int io_class_id,
...
+	if ((cache_info.info.state & (1 << ocf_cache_state_standby)) &&
+			core_id != OCF_CORE_ID_INVALID) {
+		/* Explicitly fail due to standby mode rather than
+		 * bouncing off the fact that there are 0 cores in the
+		 * cache and saying "no such core device"
+		 */
+		print_err(OCF_ERR_CACHE_STANDBY);
+		ret = FAILURE;
+		goto cleanup;
+	}
```
Comment (robertbaldyga) [inline casadm/statistics_model.c:788]: The same about io class stats.

**Review by robertbaldyga — CHANGES_REQUESTED** [review]

File: casadm/statistics_model.c [hunk 3]
```diff
@@ -632,20 +632,26 @@ static int cache_stats(int ctrl_fd, const struct kcas_cache_info *cache_info,
...
 {
 	struct kcas_get_stats cache_stats = {};
+	bool standby;
+
 	cache_stats.cache_id = cache_id;
 	cache_stats.core_id = OCF_CORE_ID_INVALID;
 	cache_stats.part_id = OCF_IO_CLASS_INVALID;
 
-	if (ioctl(ctrl_fd, KCAS_IOCTL_GET_STATS, &cache_stats) < 0)
+	standby = !!(cache_info->info.state & (1 << ocf_cache_state_standby));
```
Comment (kmajzero) [inline casadm/statistics_model.c:641]: Perhaps a function/macro similar to ocf_cache_is_standby()  would be better here? We might then refactor many other places in casadm to use such macro :)

File: casadm/statistics_model.c [hunk 4]
```diff
@@ -776,6 +776,17 @@ int cache_status(unsigned int cache_id, unsigned int core_id, int io_class_id,
 		goto cleanup;
 	}
 
+	if ((cache_info.info.state & (1 << ocf_cache_state_standby)) &&
```
Comment (kmajzero) [inline casadm/statistics_model.c:779]: Again, perhaps a function/macro similar to "ocf_cache_is_standby" would be better here?

**Review by kmajzero — CHANGES_REQUESTED** [review]

File: casadm/statistics_model.c (hunk 3 above)
Comment (arutk) [reply casadm/statistics_model.c:641]: I would happily do such refactor, but I don't have time for this at the moment. I suggest to leave it as is for now.

Comment (arutk) [reply casadm/statistics_model.c:641]: ... as this would require changing multiple lines of code

**Review by kmajzero — DISMISSED** [review]

File: casadm/statistics_model.c (hunk 2 above)
Comment (arutk) [reply casadm/statistics_model.c:790]: this is delegated to the callee function and works as expected

**Review by robertbaldyga — APPROVED** [review]

**Review by kmajzero — APPROVED** [review]

**Pull request merged**